### PR TITLE
Rework Text Field parts

### PR DESCRIPTION
**Pull request by vladmoroz**
**Breaking**
Remove the `Input` part to simplify how props are forwarded and rework internal HTML structure and styles.

**Upgrade guide**
  - All `TextField.Input` parts used **without** `TextField.Root` should be renamed to `TextField.Root`
  - All `TextField.Input` parts used **within** `TextField.Root` should be removed and their props should be put directly on the `TextField.Root` part
  - All `TextField.Slot` parts placed to the right of `TextField.Input` will need `side="right"` prop. However, no adjustment is needed when two slots were used within one `TextField`, e.g. one slot on the left and another one on the right. In that case, the slots will be automatically placed on different sides.
  - Make sure that your code works as expected if you were relying on any of the implementation quirks to override styles or behaviour.

### CODE REVIEW

File: packages/radix-ui-themes/src/components/text-field.tsx [hunk 1]
```diff
@@ -2,43 +2,55 @@
...
+    | 'hidden'
+    | 'month'
+    | 'number'
+    | 'password'
+    | 'search'
+    | 'tel'
+    | 'text'
+    | 'time'
+    | 'url'
+    | 'week';
```
Comment (vladmoroz) [inline packages/radix-ui-themes/src/components/text-field.tsx:28]: This includes every input type where you can type characters, plus `hidden`

File: packages/radix-ui-themes/src/components/text-field.css [hunk 2]
```diff
@@ -368,14 +379,15 @@
     }
     &:where(:placeholder-shown) {
       cursor: var(--cursor-disabled);
```
Comment (benoitgrelard) [inline packages/radix-ui-themes/src/components/text-field.css:381]: What's the logic behind only showing the disabled cursor if it's a placeholder and not if there's a value in the field?

File: packages/radix-ui-themes/src/components/text-field.css [hunk 3]
```diff
@@ -120,10 +120,22 @@
   &:where(:empty) {
     display: none;
   }
-  &:where(:first-child) {
+
+  /* A slot that is not a right-side slot goes on the left */
+  &:where(:not([data-side='right'])) {
+    order: -1;
     margin-left: calc(var(--text-field-border-width) * -1);
+    margin-right: 0;
```
[the remark this answers is not in the document]
Comment (vladmoroz) [reply packages/radix-ui-themes/src/components/text-field.css:128]: We could drop this `margin-right: 0` but not `margin-left: 0` on the other one

File: packages/radix-ui-themes/src/components/text-field.css (hunk 2 above)
Comment (vladmoroz) [reply packages/radix-ui-themes/src/components/text-field.css:381]: If there's a value in the field, you can still select and copy it, but when there's a placeholder, there's no interaction you can do.

**Review by benoitgrelard — APPROVED** [review]

**Pull request merged**